### PR TITLE
feat: cleanup, fix several bugs, add CI/CD

### PR DESCRIPTION
**Pull request by luwes**
I updated playerx a while ago to use standalone custom video elements for loading the different types
https://github.com/luwes/playerx/blob/main/src/playerx/options.js#L7-L13

I tried to use `hls-video-element` but it didn't work so I created a fork fixing the bugs and added a bunch of changes.
Not sure if it should be merged as is but maybe good for inspiration.

https://github.com/muxinc/youtube-video-element is also broken at the moment, working fork here https://github.com/luwes/youtube-video-element

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -31,7 +26,7 @@ Note the `type="module"`, that's important.
 
 ```html
 <head>
-  <script type="module" src="https://unpkg.com/hls-video-element@0.0"></script>
+  <script type="module" src="https://unpkg.com/hls-video-element@0"></script>
```
Comment (heff) [inline README.md:29]: When we're in the v0 realm we should pin people to specific minor versions, so we can have breaking changes in the minor version with less risk of breaking people. That's at least what we've been doing with media chrome.

File: src/hls-video-element.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
+import { SuperVideoElement } from 'super-media-element';
+import Hls from 'hls.js';
+
+class HlsVideoElement extends SuperVideoElement {
+  async attributeChangedCallback(attrName, oldValue, newValue) {
+    switch (attrName) {
+      case 'src': {
+        this.load();
```
Comment (heff) [inline src/hls-video-element.js:8]: The video element doesn't automatically load a source, so I'm not a fan of adding a difference in the API at this level.

File: src/hls-video-element.js [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+    // Wait 1 tick to allow other attributes to be set.
+    await Promise.resolve();
+
+    if (!this.src) {
+      return;
+    }
+
+    if (Hls.isSupported()) {
+      this.api = new Hls();
```
Comment (heff) [inline src/hls-video-element.js:48]: I like exposing the Hls.js API like this. 👍 Ideally we can expose more things via media-ui-extensions over time so you don't need it so much. I kind of like `api` as the name, but also this isn't the only API. Is the hope to standardize this property between custom media elements?

File: src/hls-video-element.js [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+          const loadSourceOnPlay = () => this.api.loadSource(this.src);
+          this.nativeEl.addEventListener('play', loadSourceOnPlay, {
+            once: true,
+          });
+          this.api.on(Hls.Events.DESTROYING, () => {
+            this.nativeEl.removeEventListener('play', loadSourceOnPlay);
+          });
+          break;
+        }
+        case 'metadata': {
```
Comment (heff) [inline src/hls-video-element.js:62]: Ha, making `metadata` is very cool. 👍

**Review by heff — COMMENTED** [review]
A few thoughts but the general direction looks good. Not totally clear yet why this fixes the media-playlist issue.

File: README.md (hunk 1 above)
Comment (luwes) [reply README.md:29]: good idea, will change it back

File: src/hls-video-element.js (hunk 3 above)
Comment (luwes) [reply src/hls-video-element.js:48]: yes, I've been using it for playerx to access the internal API's of the different types of video API's.
a bit like nativeEl but for Vimeo, YT, Wistia this accessed their custom API's.

**Comment by luwes** [comment]
going to close this one, it's too many changes at once. will cherry pick to fix the load issue for media-playlist

**Pull request closed without merging**